### DISCUSSION TITLE
Annotate Maxima in IPDFs

### DISCUSSION THREAD

**Discussion by Debo-rah** [mtex-toolbox/mtex#867] [Ask Anything]
Hi all!

I want to annotate the orientations that are the most present in my IPDFs. I manage to annotate specific Millerindices, however, I have a lot of datasets and often I dont know yet which orientations will be dominating. So i dont know before plotting which planes i need to annotate.

E.g. in this image, I would like to annotate the centers of the 3 red regions... So, I would need to access the values themselves, to calculate three maxima /clusters, finally find the miller indeces of them...? 
![image](https://user-images.githubusercontent.com/72927930/111666638-3089e900-8814-11eb-975d-2253ffc190d9.png)

If any one has a hint I would be very glad!
Thanks,
Deborah

**Comment by ralfHielscher** [comment]
Assuming you have given an odf you can do
```matlab
ipdf = calcPDF(odf,[],vector3d.Z)
[~,m] = max(ipdf,'numLocal',3)
plotIPDF(odf,vector3d.Z)
annotate(m,'symmetrised')
```
Ralf.

changed this according to the comment by Rüdiger

**Reply by Debo-rah** [reply]
Hi Ralf,

I tried your solution. I do not have odfs, I have ebsd orientations, but I used 

`odf = calcDensity(ebsd.orientations);`

before your code

However, i always get these two errors:
Undefined function or variable 'calcIPDF'.
Undefined function 'calcIPDF' for input arguments of type 'ODF'.

I used Mtex 5.6.0 , so the latest version.

**Reply by kilir** [reply]
Hi,
I think  `ipdf = calcPDF(odf,[],zvector)` should work.

Cheers,
Rüdiger
